Buttplug.Server.Managers.ErosTek/ET312Device.cs

```diff
@@ -0,0 +1,398 @@
+                WriteLCD("----------------", 64);
+            }
+            catch
+            {
+                // If anything goes wrong during the setup
+                // consider the entire handshake failed
+                throw new ET312HandshakeException();
+            }
+
+            // We're now ready to receive events
+            MsgFuncs.Add(typeof(FleshlightLaunchFW12Cmd), HandleFleshlightLaunchFW12Cmd);
+            MsgFuncs.Add(typeof(StopDeviceCmd), HandleStopDeviceCmd);
+
+            // Start update timer
+            updateInterval = 20;                        // <- Change this value to adjust box update frequency in ms
+            updateTimer = new Timer();
+            updateTimer.Interval = updateInterval;
+            updateTimer.Elapsed += OnUpdate;
+            updateTimer.AutoReset = true;
+            updateTimer.Enabled = true;
+        }
+
+        public override void Disconnect()
+        {
+            throw new NotImplementedException();
+        }
+
+        // Timer event fired every (updateInterval) milliseconds
+        private void OnUpdate(object source, System.Timers.ElapsedEventArgs e)
+        {
+            lock (movementLock)
+            {
+                try
+                {
+                    if (currentPosition < position)
+                    {
+                        FadeUp();
+                        currentPosition += increment;
+                        currentPosition = (currentPosition > position) ? position : currentPosition;
+                    }
+                    else if (currentPosition > position)
+                    {
+                        FadeUp();
+                        currentPosition -= increment;
+                        currentPosition = (currentPosition < position) ? position : currentPosition;
+                    }
+                    else if (currentPosition == position)
+                    {
+                        FadeDown();
+                    }
+
+                    // This is a very experimental algorithm to convert the linear "stroke"
+                    // position into the very nonlinear value the ET312 needs in order
+                    // to create a pleasant sensation
+                    double valueA = 115 + (80 * fade) + (currentPosition * 64 / 100);
+                    double valueB = 115 + (80 * fade) + ((100 - currentPosition) * 64 / 100);
+
+                    double gamma = 1.5;
+
+                    double correctedA = 255 * Math.Pow(valueA / 255, 1 / gamma);
+                    double correctedB = 255 * Math.Pow(valueB / 255, 1 / gamma);
+
+                    if (fade == 0)
+                    {
+                        correctedA = correctedB = 0;
+                    }
+
+                    Poke(0x040a5, (byte)correctedA);          // Channel A: Set intensity value
+                    Poke(0x041a5, (byte)correctedB);          // Channel B: Set intensity value
+                }
+                catch
```

Comment:
Don't blindly "catch all" only catch exceptions you're aware of and either throw ones you can't handle or don't catcch them at all.

```C#
catch(Exception ex)
{
    AbandonShip();
    if (ex is ET312Exception)
        return;
    
    // Throw the exception and preserve the stack trace
    throw;
}
```